lib/pure/concurrency/threadpool.nim

```diff
@@ -52,16 +52,14 @@ proc signal(cv: var Semaphore) =
   release(cv.L)
   signal(cv.c)
 
-const CacheLineSize = 32 # true for most archs
+const CacheLineSize = 64 # true for most archs
 
 type
   Barrier {.compilerproc.} = object
     entered: int
     cv: Semaphore # Semaphore takes 3 words at least
-    when sizeof(int) < 8:
-      cacheAlign: array[CacheLineSize-4*sizeof(int), byte]
-    left: int
-    cacheAlign2: array[CacheLineSize-sizeof(int), byte]
+    left {.align(CacheLineSize).}: int
+    pad: array[CacheLineSize-sizeof(int), byte]
     interest: bool # whether the master is interested in the "all done" event
```

Comment:
Why not `interest {.align(CacheLineSize).}: bool` and then avoid the pad/cacheAlign2 field.

Reply:
Yeah, they should be the same